product_packaging_type/models/product_packaging_type.py

```diff
@@ -138,7 +143,15 @@ def name_get(self):
         result = []
         for record in self:
             if record.product_id and record.packaging_type_id:
-                result.append((record.id, record.packaging_type_id.display_name))
+                if record.packaging_type_id.display_qty_in_name:
+                    result.append((record.id, "{} ({})".format(
+                        record.packaging_type_id.display_name,
+                        str(record.product_uom_id._compute_quantity(
+                            record.qty,
+                            record.product_uom_id)) + record.product_uom_id.name
+                    )))
+                else:
+                    result.append((record.id, record.packaging_type_id.display_name))
```

Comment:
Maybe more elegant way would be to modify the value you want to use inside append()  within if-else block (in both cases you use record.packaging_type_id.display_name alone or with qty data) and do append only once with final value at the end.